packages/blockfrost/src/blockfrostAssetProvider.ts

```diff
@@ -41,26 +41,64 @@ export const blockfrostAssetProvider = (blockfrost: BlockFrostAPI): AssetProvide
         }))
     });
 
-  const getAsset: AssetProvider['getAsset'] = async (assetId) => {
+  const getLastMintedTx = async (assetId: Cardano.AssetId): Promise<Responses['asset_history'][number] | undefined> => {
+    const [lastMintedTx] = await fetchSequentially({
+      arg: assetId.toString(),
+      haveEnoughItems: (items: Responses['asset_history']): boolean => items.length > 0,
+      paginationOptions: { order: 'desc' },
+      request: blockfrost.assetsHistory.bind<BlockFrostAPI['assetsHistory']>(blockfrost),
+      responseTranslator: (response): Responses['asset_history'] => response.filter((tx) => tx.action === 'minted')
+    });
+
+    if (!lastMintedTx) return undefined;
+    return lastMintedTx;
+  };
+
+  const getNftMetadata = async (
+    asset: Pick<Asset.AssetInfo, 'name' | 'policyId'>,
+    lastMintedTxHash: string
+  ): Promise<Asset.NftMetadata | undefined> => {
+    const metadata = await blockfrost.txsMetadata(lastMintedTxHash);
+    // Not sure if types are correct, missing 'label', but it's present in docs
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const metadatumMap = blockfrostMetadataToTxMetadata(metadata as any);
+    return Asset.util.metadatumToCip25(asset, metadatumMap, console);
+  };
+
+  const getAsset: AssetProvider['getAsset'] = async (assetId, extraData) => {
     const response = await blockfrost.assetsById(assetId.toString());
     const name = Asset.util.assetNameFromAssetId(assetId);
+    const policyId = Cardano.PolicyId(response.policy_id);
     const quantity = BigInt(response.quantity);
+    const history = async () =>
+      response.mint_or_burn_count === 1
+        ? [
+            {
+              quantity,
+              transactionId: Cardano.TransactionId(response.initial_mint_tx_hash)
+            }
+          ]
+        : await getAssetHistory(assetId);
+
+    const nftMetadata = async () => {
+      let lastMintedTxHash: string = response.initial_mint_tx_hash;
+      if (response.mint_or_burn_count > 1) {
+        const lastMintedTx = await getLastMintedTx(assetId);
+        if (lastMintedTx) lastMintedTxHash = lastMintedTx.tx_hash;
+      }
+      return getNftMetadata({ name, policyId }, lastMintedTxHash);
+    };
+
     return {
       assetId,
       fingerprint: Cardano.AssetFingerprint(response.fingerprint),
-      history:
-        response.mint_or_burn_count === 1
-          ? [
-              {
-                quantity,
-                transactionId: Cardano.TransactionId(response.initial_mint_tx_hash)
-              }
-            ]
-          : await getAssetHistory(assetId),
+      history: extraData?.history ? await history() : undefined,
+      mintOrBurnCount: response.mint_or_burn_count,
       name,
-      policyId: Cardano.PolicyId(response.policy_id),
+      nftMetadata: extraData?.nftMetadata ? await nftMetadata() : undefined,
```

Comment:
Let's change AssetInfo type's `nftMetadata` and `tokenMetadata` slightly: `nftMetadata?: NftMetadata | null` and add a doc comment like: "`undefined` if not loaded, `null` if no metadata found"

Reply:
Done